src/wifi/model/periodic-dmg-wifi-scheduler.cc

```diff
@@ -300,31 +292,29 @@ PeriodicDmgWifiScheduler::UpdateAvailableSlots (uint32_t startAlloc, uint32_t en
   uint32_t startSlot;
   uint32_t endSlot;
 
-  for (auto it = m_availableSlots.begin(); it != m_availableSlots.end(); ++it)
+  for (auto it = m_availableSlots.begin (); it != m_availableSlots.end (); ++it)
     {
-      startSlot = (*it).first;
-      endSlot = (*it).second;
-      
+      startSlot = it->first;
+      endSlot = it->second;
+
       if (startSlot > endAlloc || endSlot < startAlloc)
         {
           continue;
         }
 
       if (startSlot == startAlloc)
         {
-          m_availableSlots.insert(it, std::make_pair (endAlloc, endSlot));
-          it = m_availableSlots.erase(it);
+          it->first = endAlloc;
           break;
         }
       else if (startSlot < startAlloc && endSlot > endAlloc)
         {
-          m_availableSlots.insert(it, std::make_pair (startSlot, startAlloc));
-          m_availableSlots.insert(it, std::make_pair (endAlloc, endSlot));
-          it = m_availableSlots.erase(it);
+          it->first = endAlloc;
+          m_availableSlots.insert (it, std::make_pair (startSlot, startAlloc));
           break;
         }
```

Comment:
what if `endSlot == endAlloc`? It would probably enter the last condition and create a zero-duration slot, right?

Reply:
also, just to be sure i would add an `else` with fatal error. this holds for all if statements, especially when they are long and complicated, so it's possible that (i) you miss one edge case, (ii) some runtime error breaks one of your assumptions